### ISSUE TITLE
Creating identity with `certificate_authority` auth method fails

### ISSUE THREAD

**Issue by trdvangraft** [ansible/ansible_collections_azure#1276]
##### SUMMARY

When trying to create a device identity using the `certificate_authority` as auth method the device does not get created.
This seems to be a missing attribute within the azure_rm_iotdevice.py module. 

Checking the code it is logical as the attribute does not actually exists therefore the fix should be quite trivial
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

azure.azcollection

##### ANSIBLE VERSION

```paste below
ansible [core 2.13.12]
```

##### COLLECTION VERSION

```paste below
azure.azcollection 1.18.1
community.general  7.4.0  
```

##### OS / ENVIRONMENT

Ubuntu 20.04

##### STEPS TO REPRODUCE

```yaml
azure.azcollection.azure_rm_iotdevice:
      edge_enabled: yes
      primary_thumbprint: "{{ cert_fingerprint }}"
      secondary_thumbprint: "{{ cert_fingerprint }}"
      hub: "{{ iothub_hostname }}"
      name: "{{inventory_hostname}}"
      client_id: "{{ azure_client_id }}"
      secret: "{{ azure_secret }}"
      subscription_id: "{{ azure_subscription_id }}" 
      tenant: "{{ tenant_id }}"
      auth_method: "certificate_authority"
      hub_policy_key: {{ hub_policy_key }}
      hub_policy_name: "iothubowner"
    register: iotdevice_upload
```

##### EXPECTED RESULTS

The device gets created in the expected IoT hub

##### ACTUAL RESULTS

```paste below
FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_azure.azcollection.azure_rm_iotdevice_payload__hczpyi4/ansible_azure.azcollection.azure_rm_iotdevice_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_iotdevice.py\", line 397, in create_device\nAttributeError: 'AzureRMIoTDevice' object has no attribute 'primary_thumbprint'
```

**Comment by trdvangraft** [comment]
If required I can make a pull request where the attribute is added :)

**Comment by Fred-sun** [comment]
@trdvangraft We are very sorry for the trouble caused to you, and we have changed it, please refer to PR #1278 , thank you!

**Comment by trdvangraft** [comment]
Thank you for the quick patch :) looking forward to the release

**Comment by ggermis** [comment]
Aren't the methods also switched in the if statements in both the `create_device` and `update_device` methods? I was unable to add a `self_signed` device until I changed the type to `certificate_authority`

```
elif self.auth_method == 'self_signed':
    response = self.mgmt_client.create_device_with_certificate_authority(self.name, self.status, iot_edge=self.edge_enabled)
elif self.auth_method == 'certificate_authority':
    response = self.mgmt_client.create_device_with_x509(self.name,
                                                        self.primary_thumbprint, self.secondary_thumbprint, self.status, iot_edge=self.edge_enabled)
```

* `self_signed` should call `create_device_with_x509` 
* `certificate_authority` should call `create_device_with_certificate_authority`

**Comment by Fred-sun** [comment]
@ggermis Yes, You are right!, I will update it! Thank you very much!

**Comment by Fred-sun** [comment]
fixes by #1278
